Backport Swarm and Kubernetes deployment docs for UCP 3.0

bermudezmt opened the pull request:
Signed-off-by: Maria Bermudez <bermudez.mt@gmail.com>
### Proposed changes

This addresses #8180 and backports the following content:
- UCP architecture (removes silly redirect to 3.1 version of content)
- Adds Swarm (including interlock) and Kubernetes sections 
- Updates image links based on new subdirectories

### Unreleased project version (optional)

UCP 3.0

### Related issues (optional)

Closes #8180

bermudezmt commented:
@trapier Did a combination of awk, sed, and YAML linting for this one. Can you please review for sanity check?

GordonTheTurtle commented:
Deploy preview for *docsdocker* ready!

Built with commit 317ce6dbd56309e1d8d61a18e9f4b1f69c68445e

https://deploy-preview-8233--docsdocker.netlify.com

bermudezmt commented:
Thank you so much for previewing the PR changes, @ollypom . I hope to never do a huge backport like this again. I've fixed the redirects and updated the release notes links. This is now a dupe of #8360
